src/pages/tasks/TaskAssigneeSelectorModal.js

```diff
@@ -183,7 +183,11 @@ function TaskAssigneeSelectorModal(props) {
                 <>
                     <HeaderWithBackButton
                         title={props.translate('task.assignee')}
-                        onBackButtonPress={() => (lodashGet(props.task.report, 'isExistingTaskReport') ? Navigation.dismissModal() : Navigation.goBack(ROUTES.NEW_TASK))}
+                        onBackButtonPress={() =>
+                            props.route.params.reportID && lodashGet(props.reports, `report_${props.route.params.reportID}`, undefined)
```

Comment:
This code is duplicated. Please make common function for this

Reply:
Also use `${ONYXKEYS.COLLECTION.REPORT}${props.route.params.reportID}` instead of `report_${props.route.params.reportID}`